### PR TITLE
[Rust] Refactored the Value enum and the relevant APIs

### LINKED ISSUE #707: [Rust SDK] WasmEdge_ValueGetV128

### LINKED ISSUE #714: [Rust SDK] WasmEdge_ValueGetI64

### LINKED ISSUE #715: [Rust SDK] WasmEdge_ValueGetI32

### LINKED ISSUE #717: [Rust SDK] WasmEdge_ValueGetFuncIdx

### LINKED ISSUE #719: [Rust SDK] WasmEdge_ValueGetF64

### LINKED ISSUE #720: [Rust SDK] WasmEdge_ValueGetF32

### LINKED ISSUE #722: [Rust SDK] WasmEdge_ValueGetExternRef

### LINKED ISSUE #723: [Rust SDK] WasmEdge_ValueGenV128

### LINKED ISSUE #728: [Rust SDK] WasmEdge_ValueGenNullRef

### LINKED ISSUE #730: [Rust SDK] WasmEdge_ValueGenI64

### LINKED ISSUE #732: [Rust SDK] WasmEdge_ValueGenI32

### LINKED ISSUE #734: [Rust SDK] WasmEdge_ValueGenFuncRef

### LINKED ISSUE #738: [Rust SDK] WasmEdge_ValueGenF32

### LINKED ISSUE #739: [Rust SDK] WasmEdge_ValueGenF64

### PR DESCRIPTION
**Pull request by apepkuss**
This PR refactored the `Value` enum and the relevant APIs. The following C APIs are involved:

- Fix #732 
- Fix #730
- Fix #738 
- Fix #739 
- Fix #723 
- Fix #734 
- Fix #728 
- Fix #715 
- Fix #714 
- Fix #720 
- Fix #719 
- Fix #707 
- Fix #717
- Fix #722

### CODE REVIEW

**Comment by apepkuss** [comment]
Hi @hydai, could you please help review this PR? Thanks a lot!

File: bindings/rust/wasmedge-sys/src/types.rs [hunk 1]
```diff
@@ -278,20 +278,29 @@ pub enum Value {
...
+            wasmedge::WasmEdge_ValueGenExternRef(ptr).into()
         }
     }
 
+    // pub fn gen_extern_ref(&mut self) -> Value {
+    //     unsafe {
+    //         let self_ptr: *mut c_void = self as *mut _ as *mut c_void;
+    //         wasmedge::WasmEdge_ValueGenExternRef(self_ptr).into()
+    //     }
+    // }
```
Comment (hydai) [inline bindings/rust/wasmedge-sys/src/types.rs:302]: Why do we need to keep these comments?

**Review by hydai — CHANGES_REQUESTED** [review]

File: bindings/rust/wasmedge-sys/src/types.rs (hunk 1 above)
Comment (apepkuss) [reply bindings/rust/wasmedge-sys/src/types.rs:302]: The function has been replaced by a new one. I'll remove these lines. Thanks!

Comment (apepkuss) [reply bindings/rust/wasmedge-sys/src/types.rs]: The issue has been fixed, please check again. Thanks!

**Review by hydai — APPROVED** [review]

**Comment by hydai** [comment]
This is merged: https://github.com/WasmEdge/WasmEdge/commit/1c0d0042847c493b02af42292f87b99e497bc68b

However, the GitHub doesn't mark it resolved. So I am closing this manually.

**Pull request closed without merging**